test/action.spec.js

```diff
@@ -0,0 +1,112 @@
+const chai = require('chai');
+const expect = chai.expect;
+const sinon = require('sinon');
+const mock = require('mock-require');
+const log = require('npmlog');
+const path = require('path');
+
+const action = (config, args) => require('../src/action')(config, args || []);
+
+log.level = 'silent';
+
+describe('action', () => {
+
+  beforeEach(() => {
+    delete require.cache[require.resolve('../src/action')];
```

Comment:
Have you seen line `4` -> `const mock = require('mock-require');` ? We are doing all the things in the same way. It's just that since require calls in `action.js` are top-level ones and cached after module is required for the first time, I clean up that cache in `beforeEach` so requireMock can set them up (require mock cannot mock anything if it's been already required and put into cache).